### PR TITLE
Collection items actions selectors

### PR DESCRIPTION
**Pull request by ro-savage**
This rewrite collections to function better with our methodology
- Changes routes to be `collection-item` and `collection/id/`
- Adds sorting option using reselect
- Updates reducer names

### CODE REVIEW

File: src/views/collectionItems/collectionItems/CollectionItemsReducer.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if (orderedBy === 'default') return defaultOrder
+
+    // Options 1
+    // const listOfObjects = Object.keys(collectionItems).map(id => collectionItems[id])
+    // const sortedListOfObjects = _sortBy(listOfObjects, 'title')
+    // const sortedList = []
+    // sortedListOfObjects.forEach((item) => sortedList.push(item.id))
+
+    // Faster option, assumes we have a list of all items already.
+    const sortedObjects = _sortBy(defaultOrder, (id) => collectionItems[id][orderedBy])
```
Comment (ro-savage) [inline src/views/collectionItems/collectionItems/CollectionItemsReducer.js:61]: Uses lodash sortBy. It gets the list of objects from the defaultOrder array, then it uses those ids to read the objects, check the property stored in the var `orderedBy`, then returns an array in the new order. Its kind of hard to understand by looking at it.

Thats why i wrote it the commented out way first.

**Comment by ro-savage** [comment]
@nickmask Read for code review

File: src/views/collectionItems/collectionItems/CollectionItemsReducer.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    console.log('sortedObjects', sortedObjects)
```
Comment (nickmask) [inline src/views/collectionItems/collectionItems/CollectionItemsReducer.js]: Does this still need to be here?

**Comment by nickmask** [comment]
@ro-savage Looks good! Just a stray console log. PR approved.

**Pull request merged**